core/src/kura.rs

```diff
@@ -137,15 +147,27 @@ impl Kura {
             block_data_buffer.resize(block_data_buffer_len, 0_u8);
 
             match block_store.read_block_data(block_start, &mut block_data_buffer) {
-                Ok(_) => match VersionedCommittedBlock::decode_versioned(&block_data_buffer) {
-                    Ok(decoded_block) => {
-                        blocks.push(decoded_block);
-                    }
-                    Err(error) => {
-                        error!(?error, "Encountered malformed block. Not reading any blocks beyond this height.");
-                        break;
-                    }
-                },
+                Ok(_) => {
+                    let decoded_block = if block_stored_as_plain_text {
+                        match serde_json::from_slice::<crate::CommittedBlock>(&block_data_buffer) {
```

Comment:
```suggestion
                        match serde_json::from_slice::<crate::VersionedCommittedBlock>(&block_data_buffer) {
```

shouldn't we store blocks as versioned? as we do for the binary format. What's the reasoning here?

Reply:
as far as I see you're serializing it as versioned

Reply:
Very good of you to catch me on this. For some reason I thought it would be harder, but of course I just had to an extra derive.

Reply:
Switching to always storing and loading versioned.